File: chaincode/simple-java/src/main/java/io/bezant/SimpleChaincode.java
package io.bezant;

import org.apache.commons.logging.Log;
import org.apache.commons.logging.LogFactory;
import org.hyperledger.fabric.shim.ChaincodeBase;
import org.hyperledger.fabric.shim.ChaincodeStub;

import java.util.List;

import static java.nio.charset.StandardCharsets.UTF_8;

public class SimpleChaincode extends ChaincodeBase {
    private static final Log log = LogFactory.getLog(SimpleChaincode.class);

    public static void main(String[] args) {
        new SimpleChaincode().start(args);
    }

    @Override
    public Response init(ChaincodeStub stub) {
        log.info("========= Init =========");
        return newSuccessResponse();
    }

    @Override
    public Response invoke(ChaincodeStub stub) {
        log.info("========= Invoke =========");
        try {
            String func = stub.getFunction();
            List<String> args = stub.getParameters();

            switch (func) {
                case "get" : return get(stub, args);
                case "put" : return put(stub, args);
                case "putAndGetEnrollmentId" : return putAndGetEnrollmentId(stub, args);
                default: return newErrorResponse("No function name :" + func + " found");
            }
        }
        catch (Exception e) {
            return newErrorResponse(e.getMessage());
        }
    }

    private Response put(ChaincodeStub stub, List<String> args) {
        if (args.size() != 2) {
            return newErrorResponse("Incorrect number of arguments. Expecting 2");
        }

        stub.putStringState(args.get(0), args.get(1));

        return newSuccessResponse();
    }

    private Response get(ChaincodeStub stub, List<String> args) {
        if (args.size() != 1) {
            return newErrorResponse("Incorrect number of arguments. Expecting 1");
        }

        byte[] resultValueBytes = stub.getState(args.get(0));

        if (resultValueBytes.length == 0) {
            return newErrorResponse("Failed to get state for " + args.get(0));
        }

        return newSuccessResponse(resultValueBytes);
    }

    private Response putAndGetEnrollmentId(ChaincodeStub stub, List<String> args) {
        if (args.size() != 2) {
            return newErrorResponse("Incorrect number of arguments. Expecting 2");
        }

        stub.putStringState(args.get(0), args.get(1));

        String enrollmentId = ChaincodeUtil.getEnrollmentID(stub);
        return newSuccessResponse(enrollmentId.getBytes(UTF_8));
    }
}

File: chaincode/simple-node/simpleChaincode.js
const shim = require('fabric-shim');
const chaincodeUtil = require('./chaincodeUtil');

const SimpleChaincode = class {
    async Init(stub) {
        console.info('========= Init =========');
        return shim.success();
    }

    async Invoke(stub) {
        console.info('========= Invoke =========');
        const ret = stub.getFunctionAndParameters();
        const func = this[ret.fcn];
        if (!func) {
            return shim.error('No function name :' + ret.fcn + ' found');
        }
        try {
            return await func(stub, ret.params);
        } catch (err) {
            return shim.error(err);
        }
    }

    async put(stub, args) {
        if (args.length !== 2) {
            return shim.error('Incorrect number of arguments. Expecting 2');
        }

        const key = args[0],
            value = args[1];

        await stub.putState(key, Buffer.from(value));
        return shim.success();
    }

    async get(stub, args) {
        if (args.length !== 1) {
            return shim.error('Incorrect number of arguments. Expecting 1');
        }

        const resultValueBytes = await stub.getState(args[0]);

        if (resultValueBytes.length === 0) {
            return shim.error('Failed to get state for ' + args[0]);
        }

        return shim.success(resultValueBytes);
    }

    async putAndGetEnrollmentId(stub, args) {
        if (args.length !== 2) {
            return shim.error('Incorrect number of arguments. Expecting 2');
        }

        const key = args[0],
            value = args[1];

        await stub.putState(key, Buffer.from(value));

        const enrollmentId = chaincodeUtil.getEnrollmentId(stub);
        return shim.success(Buffer.from(enrollmentId));
    }
};

shim.start(new SimpleChaincode());